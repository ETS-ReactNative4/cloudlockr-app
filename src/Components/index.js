export { default as Brand } from './Brand'
export { default as InputField } from './InputField'
export { default as HorizontalLine } from './HorizontalLine'
export { default as Button } from './Button'
export { default as BasicHeader } from './BasicHeader'
export { default as DashboardHeader } from './DashboardHeader'
export { default as FileList } from './FileList'
export { default as DownloadDetail } from './DownloadDetail'
export { default as UploadDetail } from './UploadDetail'
export { default as WifiList } from './WifiList'
