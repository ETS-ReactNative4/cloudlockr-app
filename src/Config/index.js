export const Config = {
  mocking: {
    apiConnection: false,
    deviceConnection: true,
  },

  api: {
    url: 'https://cloudlockr.herokuapp.com/'
  },

  device: {
    expectedBluetoothName: 'DE1_DEVICE',                      // Expected discoverable device name
    connectionTimeout: 120,        // TODO: UPDATE            // Max # of seconds to wait for response data
    refreshRate: 3,                                           // # of seconds to wait between response polling
    maxBlobSizeBytes: 5120,        // TODO: UPDATE            // Max expected size of file data in message (must be lower than device message buffer as it does not account for other message data)
    maxTransmissionAttempts: 3                                // Max # of times the phone will reattempt transmission to device if error
  },
}
