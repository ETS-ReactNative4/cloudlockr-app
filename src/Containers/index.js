export { default as IndexStartupContainer } from './Startup/Index'
export { default as LoginContainer } from './Login/Index'
export { default as DashboardContainer } from './Dashboard/Index'
export { default as SettingsContainer } from './Settings/Index'
export { default as SettingsDeviceConnectionContainer } from './SettingsDeviceConnection/Index'
export { default as SettingsDevicePasswordContainer } from './SettingsDevicePassword/Index'
export { default as RegistrationContainer } from './Registration/Index'
export { default as SettingsDeviceWifiContainer } from './SettingsDeviceWifi/Index'
export { default as SettingsConfirmationContainer } from './SettingsConfirmation/Index'
export { default as UploadDownloadProgressContainer } from './UploadDownloadProgress/Index'
