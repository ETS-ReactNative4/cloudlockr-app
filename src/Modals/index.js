export { default as DownloadDetail } from './DownloadDetail'
export { default as UploadDetail } from './UploadDetail'
