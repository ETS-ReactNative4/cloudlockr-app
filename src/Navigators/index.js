export { default as ApplicationNavigator } from './Application'
export { default as LoginNavigator } from './Login'
export { default as DashboardNavigator } from './Dashboard'
export { default as SettingsNavigator } from './Settings'
export { default as SettingsDeviceConnectionNavigator } from './SettingsDeviceConnection'
export { default as SettingsDevicePasswordNavigator } from './SettingsDevicePassword'
export { default as SettingsDeviceWifiNavigator } from './SettingsDeviceWifi'
export { default as RegistrationNavigator } from './Registration'
export { default as SettingsConfirmationNavigator } from './SettingsConfirmation'
