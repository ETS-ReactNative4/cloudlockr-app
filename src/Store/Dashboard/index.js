import { buildSlice } from '@thecodingmachine/redux-toolkit-wrapper'
import SetDownloadInfo from '@/Store/Dashboard/SetDownloadInfo'

export default buildSlice('dashboard', [SetDownloadInfo], {
  downloadInfo: {},
}).reducer
