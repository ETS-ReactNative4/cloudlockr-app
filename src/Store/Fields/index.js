import { buildSlice } from '@thecodingmachine/redux-toolkit-wrapper'
import SetField from '@/Store/Fields/SetField'
import RemoveField from '@/Store/Fields/RemoveField'

export default buildSlice('fields', [SetField, RemoveField], {
  fields: {},
}).reducer
