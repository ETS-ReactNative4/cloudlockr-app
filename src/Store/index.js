import AsyncStorage from "@react-native-async-storage/async-storage";
import { combineReducers } from "redux";
import {
  persistReducer,
  persistStore,
  FLUSH,
  REHYDRATE,
  PAUSE,
  PERSIST,
  PURGE,
  REGISTER,
} from "redux-persist";
import { configureStore } from "@reduxjs/toolkit";
import initialState from "@/Store/initialState";

import startup from "./Startup";
import intention from "./Intention";
import user from "./User";
import fileTransfer from "./FileTransfer";

const reducers = combineReducers({
  startup,
  intention,
  user,
  fileTransfer,
});

const persistConfig = {
  key: "root",
  storage: AsyncStorage,
  whitelist: [],
};

const rootReducer = (state, action) => {
  // Handle special PURGE_STORE case, reset to default (does not remove local encryptionComponents)
  if (action.type === "PURGE_STORE") {
    state = initialState;
  }

  return reducers(state, action);
};

const persistedReducer = persistReducer(persistConfig, rootReducer);

const store = configureStore({
  reducer: persistedReducer,
  middleware: (getDefaultMiddleware) => {
    const middlewares = getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
      },
    });

    if (__DEV__ && !process.env.JEST_WORKER_ID) {
      const createDebugger = require("redux-flipper").default;
      middlewares.push(createDebugger());
    }

    return middlewares;
  },
});

const persistor = persistStore(store);

const PurgeStore = () => {
  return {
    type: "PURGE_STORE",
  };
};

export { store, persistor, PurgeStore };
