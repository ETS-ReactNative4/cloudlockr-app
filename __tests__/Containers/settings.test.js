import 'react-native'
import React from 'react'
import { SettingsContainer } from '../../src/Containers'
import { render } from '@testing-library/react-native'

import { store } from '../../src/Store'
import { Provider } from 'react-redux'

it('renders and matches snapshot', () => {
  const { toJSON } = render(<Provider store={store}><SettingsContainer /></Provider>);
  expect(toJSON()).toMatchSnapshot();
})
