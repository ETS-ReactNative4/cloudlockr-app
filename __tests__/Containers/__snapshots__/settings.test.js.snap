// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "backgroundColor": "#F9F9F9",
      },
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    statusBarTranslucent={true}
    supportedOrientations={
      Array [
        "landscape",
        "portrait",
      ]
    }
    transparent={true}
    visible={false}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "flex": 1,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "backgroundColor": "rgba(0, 0, 0, 0.25)",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="white"
          hidesWhenStopped={true}
          size="large"
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {},
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "bottom": 0,
                "flex": 1,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontSize": 20,
                  "fontWeight": "bold",
                  "height": 50,
                  "top": 80,
                },
                undefined,
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
  </Modal>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
        },
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#509FA8",
        },
        Object {
          "elevation": 10,
          "shadowColor": "black",
          "shadowRadius": 4,
        },
        Object {
          "height": 65,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingLeft": 15,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 28,
              "width": 28,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "paddingLeft": 15,
          },
          Object {
            "alignItems": "center",
            "flexDirection": "row",
          },
          Object {
            "paddingBottom": 2,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#F9F9F9",
              "fontFamily": "Roboto-Light",
              "fontSize": 28.8,
              "fontWeight": "300",
              "letterSpacing": 1.085,
            },
            Object {
              "textAlign": "center",
            },
          ]
        }
      >
        MySettings
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "paddingHorizontal": 45,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginBottom": 80,
          },
          Object {
            "flexDirection": "column",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "space-between",
          },
          Object {
            "height": 150,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#212529",
              "fontFamily": "Roboto-Regular",
              "fontSize": 25,
              "fontWeight": "400",
              "letterSpacing": 1.085,
            }
          }
        >
          Hello,
        </Text>
        <Text
          style={
            Object {
              "color": "#212529",
              "fontFamily": "Roboto-Light",
              "fontSize": 25,
              "fontWeight": "300",
              "letterSpacing": 1.085,
            }
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "column",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#212529",
              "fontFamily": "Roboto-Regular",
              "fontSize": 20,
              "fontWeight": "400",
              "letterSpacing": 1.085,
            }
          }
        >
          CloudLockr Device Status:
        </Text>
        <Text
          style={
            Object {
              "color": "#212529",
              "fontFamily": "Roboto-Light",
              "fontSize": 20,
              "fontWeight": "300",
              "letterSpacing": 1.085,
            }
          }
        >
          -- Checking Status --
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "column",
          },
          Object {
            "alignItems": "center",
          },
          Object {
            "justifyContent": "space-between",
          },
          Object {
            "height": 300,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#B3C38B",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#F9F9F9",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 1.085,
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Device password settings
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#B3C38B",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#F9F9F9",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 1.085,
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Device wifi settings
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginHorizontal": 45,
              },
              Object {
                "borderBottomColor": "#DFDFDF",
                "borderBottomWidth": 2,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#509FA8",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#F9F9F9",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 1.085,
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Log out
            </Text>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "marginHorizontal": 45,
              },
              Object {
                "borderBottomColor": "#DFDFDF",
                "borderBottomWidth": 2,
              },
            ]
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "flex": 1,
              },
              Object {
                "height": "100%",
                "width": "100%",
              },
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#C70000",
                "borderRadius": 8,
                "height": 50,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#F9F9F9",
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 20,
                    "fontWeight": "bold",
                    "letterSpacing": 1.085,
                  },
                  Object {
                    "textAlign": "center",
                  },
                ]
              }
            >
              Delete account
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
